deepchecks/nlp/checks/train_test_validation/text_embeddings_drift.py

```diff
@@ -0,0 +1,145 @@
+"""Module contains the embeddings drift check."""
+
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.nlp import Context, TrainTestCheck
+from deepchecks.nlp.checks.multivariate_embeddings_drift_utils import run_multivariable_drift_for_embeddings
+from deepchecks.utils.strings import format_number
+
+__all__ = ['TextEmbeddingsDrift']
+
+
+class TextEmbeddingsDrift(TrainTestCheck):
+    """
+    Calculate drift between the entire train and test datasets using a model trained to distinguish between them.
+
+    Check fits a new model to distinguish between train and test datasets, called a Domain Classifier.
+    Once the Domain Classifier is fitted the check calculates the feature importance for the domain classifier
+    model. The result of the check is based on the AUC of the domain classifier model, and the check displays
+    the change in distribution between train and test for the top features according to the
+    calculated feature importance.
+
+    Parameters
+    ----------
+    sample_size : int , default: 10_000
+        Max number of rows to use from each dataset for the training and evaluation of the domain classifier.
+    random_state : int , default: 42
+        Random seed for the check.
+    test_size : float , default: 0.3
+        Fraction of the combined datasets to use for the evaluation of the domain classifier
+    min_meaningful_drift_score : float , default 0.05
+        Minimum drift score for displaying drift in check. Under that score, check will display "nothing found".
+    dimension_reduction_method : str , default: 'auto'
+        Dimension reduction method to use for the check. Dimension reduction is used to reduce the number of
+        embeddings dimensions in order for the domain classifier to train more efficiently on the data.
+        The 2 supported methods are PCA and UMAP. While UMAP yields better results (especially visually), it is much
+        slower than PCA.
+        Supported values:
+        - 'auto' (default): Automatically choose the best method for the data. Uses UMAP if with_display is True,
+        otherwise uses PCA for a faster calculation. Doesn't use dimension reduction at all if the number of embeddings
+        dimensions is less than 30.
+        - 'pca': Use PCA for dimension reduction.
+        - 'umap': Use UMAP for dimension reduction.
+        - 'none': Don't use dimension reduction.
+    num_samples_in_display : int , default: 500
+        Number of samples to display in the check display scatter plot.
+    """
+
+    def __init__(
+            self,
+            sample_size: int = 10_000,
+            random_state: int = 42,
+            test_size: float = 0.3,
+            min_meaningful_drift_score: float = 0.05,
+            dimension_reduction_method: str = 'auto',
+            num_samples_in_display: int = 500,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.sample_size = sample_size
```

Comment:
Follow up on @Nadav-Barak s comment - check if embeddings are calculated, and quit (error) if not.

Reply:
The .embeddings attribute throws an exception if called without embeddings, like metadata and properties. Why do anything different?

Reply:
didn't know